Crash Report FIleNotFoundException in Storage Calculator

**Describe the bug**
```
java.lang.RuntimeException: 

 
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:557)

 
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:873)
Caused by: java.lang.reflect.InvocationTargetException: 

 
  at java.lang.reflect.Method.invoke (Method.java)

 
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:547)

 
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:873)
Caused by: java.io.FileNotFoundException: 

 
  at android.os.storage.StorageManager.getUuidForPath (StorageManager.java:767)

 
  at org.kiwix.kiwixmobile.settings.StorageCalculator.a (StorageCalculator.java:14)

 
  at org.kiwix.kiwixmobile.zim_manager.library_view.LibraryFragment.getSpaceAvailable (LibraryFragment.java)

 
  at org.kiwix.kiwixmobile.zim_manager.library_view.LibraryFragment.notEnoughSpaceAvailable (LibraryFragment.java)

 
  at org.kiwix.kiwixmobile.zim_manager.library_view.LibraryFragment.onBookItemClick (LibraryFragment.java)

 
  at org.kiwix.kiwixmobile.zim_manager.library_view.LibraryFragment.access$onBookItemClick (LibraryFragment.java)

 
  at org.kiwix.kiwixmobile.zim_manager.library_view.LibraryFragment$libraryAdapter$2$1.invoke (LibraryFragment.java)

 
  at org.kiwix.kiwixmobile.zim_manager.library_view.LibraryFragment$libraryAdapter$2$1.invoke (LibraryFragment.java)

 
  at org.kiwix.kiwixmobile.zim_manager.library_view.adapter.LibraryViewHolder$LibraryBookViewHolder$bind$1.onClick (LibraryViewHolder.java)

 
  at android.view.View.performClick (View.java:6310)

 
  at android.view.View$PerformClick.run (View.java:24970)

 
  at android.os.Handler.handleCallback (Handler.java:794)

 
  at android.os.Handler.dispatchMessage (Handler.java:99)

 
  at android.os.Looper.loop (Looper.java:176)

 
  at android.app.ActivityThread.main (ActivityThread.java:6662)

 
  at java.lang.reflect.Method.invoke (Method.java)

 
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:547)

 
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:873)
```

**Expected behavior**
No crash

**Steps to reproduce the behavior:**
I believe this happens we a user has set removable storage as their preferred storage and then removes that storage

**Environment**
- Version of Kiwix Android : 3.0.1
- Device : Xiaomi Redmi 6A (cactus)
- OS version : 8.1

--- a/app/src/main/java/org/kiwix/kiwixmobile/settings/StorageCalculator.kt
+++ b/app/src/main/java/org/kiwix/kiwixmobile/settings/StorageCalculator.kt
@@ -35,10 +35,14 @@ class StorageCalculator @Inject constructor(private val storageManager: StorageM
 
   @SuppressLint("NewApi")
   fun availableBytes(file: File) =
-    if (KiwixBuildConfig.SDK_INT >= VERSION_CODES.O)
-      storageManager.getAllocatableBytes(storageManager.getUuidForPath(file))
-    else
-      file.freeSpace
+    if (file.exists()) {
+      if (KiwixBuildConfig.SDK_INT >= VERSION_CODES.O)
+        storageManager.getAllocatableBytes(storageManager.getUuidForPath(file))
+      else
+        file.freeSpace
+    } else {
+      0L
+    }
 
-  private fun totalBytes(file: File) = file.totalSpace
+  private fun totalBytes(file: File) = if (file.exists()) file.totalSpace else 0L
 }

--- a/app/src/test/java/org/kiwix/kiwixmobile/settings/StorageCalculatorTest.kt
+++ b/app/src/test/java/org/kiwix/kiwixmobile/settings/StorageCalculatorTest.kt
@@ -39,24 +39,39 @@ internal class StorageCalculatorTest {
   }
 
   @Test
-  fun calculateAvailableSpace() {
+  fun `calculate available space with existing file`() {
     every { KiwixBuildConfig.SDK_INT } returns 25
     every { file.freeSpace } returns 1
+    every { file.exists() } returns true
     assertThat(storageCalculator.calculateAvailableSpace(file)).isEqualTo("1 Bytes")
   }
 
   @Test
-  fun calculateTotalSpace() {
+  fun `calculate total space of existing file`() {
     every { file.totalSpace } returns 1
+    every { file.exists() } returns true
     assertThat(storageCalculator.calculateTotalSpace(file)).isEqualTo("1 Bytes")
   }
 
   @Test
-  fun availableBytes() {
+  fun `calculate total space of non existing file`() {
+    every { file.exists() } returns false
+    assertThat(storageCalculator.calculateTotalSpace(file)).isEqualTo("0 Bytes")
+  }
+
+  @Test
+  fun `available bytes of existing file API 26`() {
     val uuid: UUID = mockk()
     every { KiwixBuildConfig.SDK_INT } returns 26
     every { storageManager.getUuidForPath(file) } returns uuid
     every { storageManager.getAllocatableBytes(uuid) } returns 1
+    every { file.exists() } returns true
     assertThat(storageCalculator.availableBytes(file)).isEqualTo(1L)
+  }
+
+  @Test
+  fun `available bytes of non existing file`() {
+    every { file.exists() } returns false
+    assertThat(storageCalculator.availableBytes(file)).isEqualTo(0L)
   }
 }

